api/v1beta1/nodepool_types.go

```diff
@@ -655,6 +655,13 @@ type KubevirtNodePoolPlatform struct {
 	// +optional
 	// +kubebuilder:default={memory: "4Gi", cores: 2}
 	Compute *KubevirtCompute `json:"compute"`
+
+	// NetworkInterfaceMultiQueue If specified, virtual network interfaces configured with a virtio bus will also enable
+	// the vhost multiqueue feature for network devices. The number of queues created depends on additional factors of
+	// the VirtualMachineInstance, like the number of guest CPUs.
+	//
+	// +optional
+	NetworkInterfaceMultiQueue *bool `json:"networkInterfaceMultiqueue,omitempty"`
```

Comment:
The PR makes sense. I'm a little torn on whether this should be a boolean or an enum. Here's why

It's possible in the future that we might want to attempt to autodetect whether or not multiqueue should be used or not (tracking this idea [here](https://issues.redhat.com/browse/CNV-30445)). If we want something like  `networkInterfaceMultiqueue=Auto` in the future, then I think we'd want make this value an enum, with example values being something like [Disabled, Enabled, Auto]

do you think that would make sense? So we'd start with the values [Disabled, Enabled] for now, and maybe expand it in the future?

Reply:
I got your point. Just wander why it's  a boolean in kv. I followed kv api on purpose, to make it easier for the user. But if you think enum makes more sense, I'll change the api.

Reply:
the difference is, in hypershift we have discovered that multiqueue is not always beneficial due to fragmentation of double encapsulation of geneve traffic when small MTUs are in use. that's not a common scenario in standard KubeVirt VMs. we're still learning about this scenario, and i anticipate our recommendations around when to enable MQ will evolve for hypershift.

in 4.14 we're creating this toggle to enable/disable multiqueue and we'll handle communicating when to use this in the documentation.

in 4.15, we'll add the additional `Auto` value to the hypershift api for network multiqueue, which will result in us probing the infra cluster and determining whether MQ should be used or not.